### DISCUSSION TITLE
Potential Herd SSL issue (solution)

### DISCUSSION THREAD

**Discussion by alexgarrettsmith** [exposedev/expose#399] [Show and tell]
Here's a potential solution for anyone struggling with SSL errors when using Expose via Herd.

Here's the error I was getting:

```
Could not connect to the server.
Connection to tls://sharedwithexpose.com:443 failed during TLS handshake: SSL operation failed with code 1. OpenSSL Error messages: error:0A000086:SSL routines::certificate verify failed
```

I debugged this by running `expose share codecourse -vvv` and saw I was running PHP directly from the Laravel Herd binaries and the SSL configuration was missing.

**To solve this for now, I've simply installed a PHP standalone with Homebrew specifically for the terminal (`brew install php@8.2`) and it's working.**

There's probably a much better solution for this in the long term, but hope this helps for now.

**Comment by michaelklopf** [comment]
We just learned that the problem was a missing config, which is `HERD_PHP_83_INI_SCAN_DIR` or `PHP_INI_SCAN_DIR`.

This article states https://beyondco.de/blog/announcing-laravel-herd that

> When Herd gets installed, this environment variable is automatically written to your ~/.zshrc / ~/.bashrc file.
> 
> `export PHP_INI_SCAN_DIR="/Users/USERNAME/Library/Application Support/Herd/config/php/":$PHP_INI_SCAN_DIR`
> 
> As Herd is a MacOS app, and all MacOS app have a shared folder for all of their app specific files, we can just hardcode this environment variable.

But neither file has been changed to include that config and we use fish shell so completely missed it.

The docs for Herd or Expose could be clearer about that.

**Reply by n4ff4h** [reply]
thanks this worked for me

**Comment by stevepop** [comment]
Hi @alexgarrettsmith 
I followed your potential solution above but even after installing PHP with brew, the error did not go away. I can also confirm that the certificate is at the location where it is being referenced.
